context/src/bazel_bep/parser.rs

```diff
@@ -63,23 +68,27 @@ impl BazelBepParser {
         let file = std::fs::File::open(&self.bazel_bep_path)?;
         let reader = std::io::BufReader::new(file);
 
-        let (errors, test_results) = Deserializer::from_reader(reader)
+        let (errors, test_results, bep_test_events) = Deserializer::from_reader(reader)
             .into_iter::<BuildEvent>()
             .fold(
-                (Vec::<String>::new(), Vec::<TestResult>::new()),
-                |(mut errors, mut test_results), parse_event| {
+                (
+                    Vec::<String>::new(),
+                    Vec::<TestResult>::new(),
+                    Vec::<BuildEvent>::new(),
+                ),
+                |(mut errors, mut test_results, mut bep_test_events), parse_event| {
                     match parse_event {
                         Result::Err(ref err) => {
                             errors.push(format!("Error parsing build event: {}", err));
                         }
                         Result::Ok(build_event) => {
-                            if let Some(Payload::TestResult(test_result)) = build_event.payload {
-                                let xml_files = test_result
+                            if let Some(Payload::TestResult(test_result)) = &build_event.payload {
```

Comment:
I will also be uploading `TestSummary` events in a follow-up. I considered the `started` events too, but that can include some env vars and bazel config that we probably shouldn't collect